lib/units/ios-device/plugins/devicelog.js

```diff
@@ -17,79 +18,69 @@ export default syrup
-            , '--debug'
-            , '--noinstall'
-            , '--bundle'
-        ]
+
         let DeviceLogger = {
             appOptions: {}
             , channel: ''
             , stream: null
+
             , setChannel: channel => {
-                this.channel = channel
+                DeviceLogger.channel = channel
             }
-            , startLoggging: function(channel, deviceData) {
+
+            , startLoggging: async function(channel) {
                 DeviceLogger.setChannel(channel)
-                if (deviceData && deviceData.bundleName.length !== 0 &&
-                deviceData.bundleName.substr(-4) === '.app') {
-                    group.get().then(group => {
-                        launchArgs.push(deviceData.bundleName)
-                        this.stream = spawn('ios-deploy1', launchArgs, {
-                            shell: true
-                            , cwd: deviceData.pathToApp
-                        })
-                        this.stream.stdout.on('data', data => {
-                            push.send([
-                                group.group
-                                , wireutil.envelope(new wire.DeviceLogcatEntryMessage(options.serial, new Date().getTime() / 1000, this.stream.pid, this.stream.pid, 1, 'device:log:cat', data.toString()))
-                            ])
-                        })
-                        this.stream.on('close', err => {
-                            log.fatal('Unable to get devicelog', err)
-                            this.killLoggingProcess()
-                        })
+                try {
+                    DeviceLogger.stream = spawn('idb logs --udid ' + options.serial, {shell: true})
+
+                    DeviceLogger.stream.stdout.on('data', data => {
+                        push.send([
+                            channel
+                            , wireutil.envelope(new wire.DeviceLogcatEntryMessage(options.serial, new Date().getTime() / 1000, DeviceLogger.stream.pid, DeviceLogger.stream.pid, 1, 'device:log:cat', data.toString()))
+                        ])
+                    })
+
+                    DeviceLogger.stream.on('close', err => {
+                        log.fatal('Unable to get devicelog', err)
+                        DeviceLogger.killLoggingProcess()
                     })
                 }
+                catch (error) {
+                    log.error('Error starting logging', error)
+                }
             }
+
             , killLoggingProcess: () => {
-                if (this.stream) {
-                    process.kill(-this.stream.pid)
-                    this.stream.kill()
-                    this.stream = null
-                    this.channel = ''
+                if (DeviceLogger.stream) {
+                    process.kill(-DeviceLogger.stream.pid)
+                    DeviceLogger.stream.kill()
+                    DeviceLogger.stream = null
+                    DeviceLogger.channel = ''
```

Comment:
`channel` should either be cleared anyways - disregarding the stream being opened or closed, or or not cleaned at all, since it's not the same as logging process

Reply:
What if somebody does setChannel() then startLogging() then killLoggingProcess() and then startLogging() again?